Modules/_sqlite/module.c

```diff
@@ -338,10 +341,21 @@ do {                                           \
     }                                          \
 } while (0)
 
+#define ADD_EXCEPTION(module, name, exc, base)                  \
+do {                                                            \
+    exc = PyErr_NewException(MODULE_NAME "." name, base, NULL); \
+    if (!exc) {                                                 \
+        goto error;                                             \
+    }                                                           \
+    if (PyModule_AddObject(module, name, exc) < 0) {            \
```

Comment:
You may need to incref the object before using PyModule_AddObject

https://github.com/python/cpython/blob/037245c5ac46c3436f617a1f5d965929754be239/Modules/_testcapimodule.c#L7156-L7158

Reply:
```
Note

Unlike other functions that steal references, PyModule_AddObject() only decrements the reference count of value on success.

This means that its return value must be checked, and calling code must Py_DECREF() value manually on error. Example usage:
```